contracts/schemes/UpgradeScheme.sol

```diff
@@ -61,106 +73,106 @@ contract UpgradeScheme is Initializable, VotingMachineCallbacks, ProposalExecute
         avatar = _avatar;
         votingMachine = _votingMachine;
         voteParams = _voteParams;
+        package = _package;
     }
 
     /**
     * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
     * @param _proposalId the ID of the voting in the voting machine
-    * @param _param a parameter of the voting result, 1 yes and 2 is no.
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    * @return bool success
     */
-    function executeProposal(bytes32 _proposalId, int256 _param) external onlyVotingMachine(_proposalId) returns(bool) {
-        UpgradeProposal memory proposal = organizationProposals[_proposalId];
-        require(proposal.proposalType != 0);
-        delete organizationProposals[_proposalId];
-        emit ProposalDeleted(address(avatar), _proposalId);
-        // Check if vote was successful:
-        if (_param == 1) {
-
-        // Define controller and get the params:
-            Controller controller = Controller(avatar.owner());
-        // Upgrading controller:
-            if (proposal.proposalType == 1) {
-                require(controller.upgradeController(proposal.upgradeContract));
-            }
-
-        // Changing upgrade scheme:
-            if (proposal.proposalType == 2) {
-                bytes4 permissions = controller.schemesPermissions(address(this));
-                require(
-                controller.registerScheme(proposal.upgradeContract, permissions)
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        Proposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.exist, "must be a live proposal");
+        require(proposal.passed == false, "cannot execute twice");
```

Comment:
do we now need these ?

Reply:
Exists I think yes, but I removed the passed property